Source/Core/S2Cell.js

```diff
@@ -0,0 +1,589 @@
+    default:
+      return new Cartesian3(v, u, -1);
+  }
+}
+
+/**
+ * @private
+ */
+function STtoUV(s) {
+  if (s >= 0.5) return (1 / 3) * (4 * s * s - 1);
+  return (1 / 3) * (1 - 4 * (1 - s) * (1 - s));
+}
+
+/**
+ * @private
+ */
+function SiTitoST(si) {
+  return (1.0 / S2MaxSiTi) * si;
+}
+
+/**
+ * @private
+ */
+function IJLeveltoBoundUV(ij, level) {
+  var result = [[], []];
+  var cellSize = GetSizeIJ(level);
+  for (var d = 0; d < 2; ++d) {
+    var ijLo = ij[d] & -cellSize;
+    var ijHi = ijLo + cellSize;
+    result[d][0] = STtoUV(IJtoSTMin(ijLo));
+    result[d][1] = STtoUV(IJtoSTMin(ijHi));
+  }
+  return result;
+}
+
+/**
+ * @private
+ */
+function GetSizeIJ(level) {
+  return (1 << (S2MaxLevel - level)) >>> 0;
+}
+
+/**
+ * @private
+ */
+function IJtoSTMin(i) {
+  return (1.0 / S2LimitIJ) * i;
+}
+
+// Utility Functions
+
+/**
+ * @private
+ */
+function generateLookupCell(
+  level,
+  i,
+  j,
+  originalOrientation,
+  position,
+  orientation
+) {
+  if (level === S2LookupBits) {
+    var ij = (i << S2LookupBits) + j;
+    S2LookupPositions[(ij << 2) + originalOrientation] =
+      (position << 2) + orientation;
+    S2LookupIJ[(position << 2) + originalOrientation] = (ij << 2) + orientation;
+  } else {
+    level++;
+    i <<= 1;
+    j <<= 1;
+    position <<= 2;
+    var r = S2PosToIJ[orientation];
+    generateLookupCell(
+      level,
+      i + (r[0] >> 1),
+      j + (r[0] & 1),
+      originalOrientation,
+      position,
+      orientation ^ S2PosToOrientationMask[0]
+    );
+    generateLookupCell(
+      level,
+      i + (r[1] >> 1),
+      j + (r[1] & 1),
+      originalOrientation,
+      position + 1,
+      orientation ^ S2PosToOrientationMask[1]
+    );
+    generateLookupCell(
+      level,
+      i + (r[2] >> 1),
+      j + (r[2] & 1),
+      originalOrientation,
+      position + 2,
+      orientation ^ S2PosToOrientationMask[2]
+    );
+    generateLookupCell(
+      level,
+      i + (r[3] >> 1),
+      j + (r[3] & 1),
+      originalOrientation,
+      position + 3,
+      orientation ^ S2PosToOrientationMask[3]
+    );
+  }
+}
+
+/**
+ * @private
+ */
+function generateLookupTable() {
+  generateLookupCell(0, 0, 0, 0, 0, 0);
+  generateLookupCell(0, 0, 0, S2SwapMask, 0, S2SwapMask);
+  generateLookupCell(0, 0, 0, S2InvertMask, 0, S2InvertMask);
+  generateLookupCell(
+    0,
+    0,
+    0,
+    S2SwapMask | S2InvertMask,
+    0,
+    S2SwapMask | S2InvertMask
+  );
+}
+
+/**
+ * Return the lowest-numbered bit that is on for this cell id
+ * @private
+ */
+function lsb(cellId) {
+  return cellId & (~cellId + BigInt(1)); // eslint-disable-line
+}
+
+/**
+ * Return the lowest-numbered bit that is on for cells at the given level.
+ * @private
+ */
+function lsbForLevel(level) {
+  return BigInt(1) << BigInt(2 * (S2MaxLevel - level)); // eslint-disable-line
+}
+
+/**
+ * Return the number of trailing zeros in number.
+ * @private
+ */
+function countTrailingZero(x) {
```

Comment:
This function name sounds better with plural "zeros" 

```suggestion
function countTrailingZeros(x) {
```

Reply:
one alternative is `countTrailingZeroBits`